crates/bevy_ecs/src/schedule_v3/schedule.rs

```diff
@@ -0,0 +1,1001 @@
+        #[cfg(feature = "trace")]
+        let _span = bevy_utils::tracing::info_span!("check ticks").entered();
+        for schedule in self.inner.values_mut().flatten() {
+            schedule.check_change_ticks(change_tick, last_check_tick);
+        }
+    }
+}
+
+/// TBD
+pub struct Schedule {
+    graph: ScheduleMeta,
+    executable: SystemSchedule,
+    executor: Box<dyn SystemExecutor>,
+}
+
+impl Schedule {
+    pub fn new() -> Self {
+        Self {
+            graph: ScheduleMeta::new(),
+            executable: SystemSchedule::new(),
+            executor: Box::new(MultiThreadedExecutor::new()),
+        }
+    }
+
+    pub fn add_system<P>(&mut self, system: impl IntoSystemConfig<P>) {
+        self.graph.add_system(system);
+    }
+
+    pub fn add_systems<P>(&mut self, systems: impl IntoSystemConfigs<P>) {
+        self.graph.add_systems(systems);
+    }
+
+    pub fn configure_set(&mut self, set: impl IntoSystemSetConfig) {
+        self.graph.configure_set(set);
+    }
+
+    pub fn configure_sets(&mut self, sets: impl IntoSystemSetConfigs) {
+        self.graph.configure_sets(sets);
+    }
+
+    pub fn set_default_set(&mut self, set: impl SystemSet) {
+        self.graph.set_default_set(set);
+    }
+
+    pub fn set_executor(&mut self, executor: ExecutorKind) {
+        self.executor = match executor {
+            ExecutorKind::Simple => Box::new(SimpleExecutor::new()),
+            ExecutorKind::SingleThreaded => Box::new(SingleThreadedExecutor::new()),
+            ExecutorKind::MultiThreaded => Box::new(MultiThreadedExecutor::new()),
+        };
+    }
+
+    pub fn run(&mut self, world: &mut World) {
+        self.initialize(world).unwrap();
+        self.executor.run(&mut self.executable, world);
+    }
+
+    pub(crate) fn initialize(&mut self, world: &mut World) -> Result<(), BuildError> {
+        if self.graph.changed {
+            self.graph.initialize(world);
+            self.graph.update_schedule(&mut self.executable)?;
+            self.executor.init(&self.executable);
+        }
+
+        Ok(())
+    }
+
+    /// Iterates all component change ticks and clamps any older than [`MAX_CHANGE_AGE`](crate::change_detection::MAX_CHANGE_AGE).
+    /// This prevents overflow and thus prevents false positives.
+    ///
+    /// **Note:** Does nothing if the [`World`] counter has not been incremented at least [`CHECK_TICK_THRESHOLD`](crate::change_detection::CHECK_TICK_THRESHOLD)
+    /// times since the previous pass.
+    pub(crate) fn check_change_ticks(&mut self, _change_tick: u32, _last_change_tick: u32) {
+        #[cfg(feature = "trace")]
+        let _span = bevy_utils::tracing::info_span!("check schedule ticks").entered();
+        todo!();
+    }
+}
+
+#[derive(Default)]
+struct DAG {
+    /// A directed graph.
+    graph: DiGraphMap<NodeId, ()>,
+    /// A cached topological ordering of the graph.
+    topsort: Vec<NodeId>,
+}
+
+impl DAG {
+    pub fn new() -> Self {
```

Comment:
```suggestion
    fn new() -> Self {
```

No point making this pub when the struct isn't.